lib/epochs.js

```diff
@@ -78,27 +78,72 @@ module.exports = function Epochs(ssb) {
     },
   }
 
+  function getEpochs(groupId, cb) {
+    if (cb === undefined) return p(this.getEpochs)(groupId)
+
+    const getters = pluck(allGetters, ['author', 'secret'])
+    reduceEpochs(ssb, groupId, getters, (err, reduce) => {
+      if (err) return cb(clarify(err, 'Failed to resolve epoch @tangle/reduce'))
+
+      const epochs = reduce.graph.connectedNodes.map((node) => {
+        const info = {
+          id: node.key, // alias: epochRootId
+          previous: node.previous,
+          ...node.data[node.key],
+        }
+        info.secret = Buffer.from(info.secret, 'base64')
+        return info
+      })
+
+      cb(null, epochs)
+    })
+  }
+
   return {
-    getEpochs(groupId, cb) {
-      if (cb === undefined) return p(this.getEpochs)(groupId)
-
-      const getters = pluck(allGetters, ['author', 'secret'])
-      reduceEpochs(ssb, groupId, getters, (err, reduce) => {
-        if (err)
-          return cb(clarify(err, 'Failed to resolve epoch @tangle/reduce'))
-
-        const epochs = reduce.graph.connectedNodes.map((node) => {
-          const info = {
-            id: node.key, // alias: epochRootId
-            previous: node.previous,
-            ...node.data[node.key],
-          }
-          info.secret = Buffer.from(info.secret, 'base64')
-          return info
+    getEpochs,
+
+    /**
+     * Get the epoch matching the currently picked write key for the group.
+     */
+    getPickedEpoch(groupId, opts) {
```

Comment:
This function is already one I wrote in https://github.com/ssbc/ssb-tribes2/pull/99 ... :cry: 
I'm gonna get that merged and you should use that because I'm writing legit tests around it based on the specs

Reply:
:fire: TO CHANGE

Reply:
also I called it `getPreferredEpoch` to match the name in the spec